.github/workflows/CI.yml

```diff
@@ -0,0 +1,31 @@
+name: CI
+on: [push]
```

Comment:
do we want this on all push, or just pull request and push to main?

Reply:
The julia sdk workflow agrees with you, so I've joined the majority and changed it.